src/backend/src/user_profile.rs

```diff
@@ -35,3 +40,31 @@ pub fn create_profile(
         default_profile
     }
 }
+
+pub fn add_credential(
+    principal: StoredPrincipal,
+    credential_type: &CredentialType,
+    user_profile_map: &mut StableBTreeMap<(u64, StoredPrincipal), Candid<StoredUserProfile>, VMem>,
+    user_profile_updated_map: &mut StableBTreeMap<StoredPrincipal, u64, VMem>,
+) -> Result<(), AddUserCredentialError> {
+    if let Ok(user_profile) = get_profile(principal, user_profile_map, user_profile_updated_map) {
+        let now = time();
+        let user_credential = UserCredential {
+            credential_type: credential_type.clone(),
+            verified_date_timestamp: Some(now),
+        };
+        let mut credentials = user_profile.credentials.clone();
+        credentials.insert(credential_type.clone(), user_credential);
+        let updated_profile = StoredUserProfile {
+            created_timestamp: user_profile.created_timestamp,
+            updated_timestamp: now,
+            version: Some(user_profile.version.map_or(1, |v| v + 1)),
```

Comment:
Can you implement a similar trait as `TokenVersion` for user profile.

Maybe we should actually rename `TokenVersion` and reuse it for tokens and any types of entity that has a field version.

Reply:
Done!